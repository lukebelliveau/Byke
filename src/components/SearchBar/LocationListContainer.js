// @flow
import React from 'react';
import { connect } from 'react-redux';
import LocationList from './LocationList';
import actions from '../../redux/actions';

const mapDispatchToProps = dispatch => ({
  tripSet: trip => {
    dispatch(actions.tripSet(trip));
  },
});

const mapStateToProps = state => ({
  locations: state.locations,
  trip: state.trip,
});

export default connect(mapStateToProps, mapDispatchToProps)(LocationList);
