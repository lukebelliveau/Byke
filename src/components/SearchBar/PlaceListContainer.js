// @flow
import React from 'react';
import { connect } from 'react-redux';
import PlaceList from './PlaceList';
import actions from '../../redux/actions';

const mapDispatchToProps = dispatch => ({
  tripSet: trip => {
    dispatch(actions.tripSet(trip));
  },
});

const mapStateToProps = state => ({
  places: state.places,
  trip: state.trip,
});

export default connect(mapStateToProps, mapDispatchToProps)(PlaceList);
