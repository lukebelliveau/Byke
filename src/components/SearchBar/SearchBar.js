// @flow
import React, { Component } from 'react';
import {
  View,
  TextInput,
  Text,
  StyleSheet,
  Platform,
  TouchableOpacity,
} from 'react-native';

type PropTypes = {
  searchText: string,
  changeSearchText: string => void,
  searchPlaces: string => void,
};

const SearchBar = ({
  searchText,
  changeSearchText,
  searchPlaces,
  trip,
  exitTrip,
}: PropTypes) =>
  <View style={[styles.container]}>
    {trip
      ? <TouchableOpacity onPress={exitTrip} style={styles.button}>
          <Text style={styles.arrow}>{'<'}</Text>
        </TouchableOpacity>
      : null}

    <TextInput
      style={styles.input}
      underlineColorAndroid="transparent"
      placeholder="Where are you going?"
      placeholderTextColor="lightgray"
      value={searchText}
      onChangeText={changeSearchText}
      onSubmitEditing={() => searchPlaces(searchText)}
      returnKeyType="go"
      selectionColor="darkgrey"
    />
  </View>;

const primaryColor = 'dodgerblue';
const secondaryColor = '#1b81e5';
const styles = StyleSheet.create({
  button: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    backgroundColor: secondaryColor,
    width: 40,
    height: 40,
    borderRadius: 50,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  arrow: {
    fontSize: 30,
    fontWeight: 'bold',
    color: 'white',
  },
  container: {
    flex: 1,
    flexDirection: 'row',
    paddingTop: 20,
    paddingLeft: 10,
    backgroundColor: primaryColor,
    alignItems: 'center',
    shadowColor: 'black',
    shadowRadius: 10,
    shadowOpacity: 100,
    elevation: 10,
  },
  input: {
    flex: Platform.OS === 'ios' ? 6 : 0,
    maxHeight: Platform.OS === 'ios' ? 25 : 50,
    fontSize: 15,
    width: '75%',
    backgroundColor: secondaryColor,
    color: 'white',
    borderRadius: 10,
    margin: 20,
    padding: 5,
    paddingLeft: 10,
  },
});

export default SearchBar;
