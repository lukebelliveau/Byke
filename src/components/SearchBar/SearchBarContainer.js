import React from 'react';
import { connect } from 'react-redux';
import SearchBar from './SearchBar';
import actions from '../../redux/actions';

const mapDispatchToProps = dispatch => ({
  locationsFetched: locations => {
    dispatch(actions.locationsFetched(locations));
  },
  loadingStarted: () => {
    dispatch(actions.loadingStarted());
  },
  loadingFinished: () => {
    dispatch(actions.loadingFinished());
  },
  searchLocations: query => {
    dispatch(actions.searchLocations(query));
  },
});

const mapStateToProps = state => ({
  region: state.region,
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
