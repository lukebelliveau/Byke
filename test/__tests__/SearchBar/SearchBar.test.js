import React from 'react';
import { TextInput, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SearchBar from '../../../src/components/SearchBar/SearchBar';

const userSubmitted = jest.fn();
const entryString = 'DisneyWorld';

const trip = {
  destination: {
    latitude: 50,
    longitude: 50,
  },
};

it('calls userSubmitted callback when user presses submit', () => {
  const input = shallow(
    <SearchBar searchText={entryString} searchPlaces={userSubmitted} />
  ).find(TextInput);

  input.simulate('submitEditing');

  expect(userSubmitted).toBeCalledWith(entryString);
});

it('does not contain a back button when not in trip mode', () => {
  const searchBar = renderer
    .create(
      <SearchBar
        searchText={entryString}
        searchPlaces={userSubmitted}
        trip={null}
      />
    )
    .toJSON();

  expect(searchBar).toMatchSnapshot();
});

it('contains a back button when in trip mode', () => {
  const searchBar = renderer
    .create(
      <SearchBar
        searchText={entryString}
        searchPlaces={userSubmitted}
        trip={trip}
      />
    )
    .toJSON();

  expect(searchBar).toMatchSnapshot();
});

it('exits trip when back button is pressed', () => {
  const exitTrip = jest.fn();
  const searchBar = shallow(
    <SearchBar
      searchText={entryString}
      searchPlaces={userSubmitted}
      trip={trip}
      exitTrip={exitTrip}
    />
  );

  searchBar.find(TouchableOpacity).simulate('press');

  expect(exitTrip).toBeCalled();
});
