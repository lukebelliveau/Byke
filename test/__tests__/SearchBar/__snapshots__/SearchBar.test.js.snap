// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contains a back button when in trip mode 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "elevation": 10,
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingTop": 20,
        "shadowColor": "black",
        "shadowOpacity": 100,
        "shadowRadius": 10,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1b81e5",
        "borderRadius": 50,
        "display": "flex",
        "flex": 1,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      &lt;
    </Text>
  </View>
  <TextInput
    onChangeText={undefined}
    onSubmitEditing={[Function]}
    placeholder="Where are you going?"
    placeholderTextColor="lightgray"
    returnKeyType="go"
    selectionColor="darkgrey"
    style={
      Object {
        "backgroundColor": "#1b81e5",
        "borderRadius": 10,
        "color": "white",
        "flex": 6,
        "fontSize": 15,
        "margin": 20,
        "maxHeight": 25,
        "padding": 5,
        "paddingLeft": 10,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value="DisneyWorld"
  />
</View>
`;

exports[`does not contain a back button when not in trip mode 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "elevation": 10,
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingTop": 20,
        "shadowColor": "black",
        "shadowOpacity": 100,
        "shadowRadius": 10,
      },
    ]
  }
>
  <TextInput
    onChangeText={undefined}
    onSubmitEditing={[Function]}
    placeholder="Where are you going?"
    placeholderTextColor="lightgray"
    returnKeyType="go"
    selectionColor="darkgrey"
    style={
      Object {
        "backgroundColor": "#1b81e5",
        "borderRadius": 10,
        "color": "white",
        "flex": 6,
        "fontSize": 15,
        "margin": 20,
        "maxHeight": 25,
        "padding": 5,
        "paddingLeft": 10,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value="DisneyWorld"
  />
</View>
`;
